Foundation/NSKeyedArchiver.swift

```diff
@@ -132,20 +132,21 @@ open class NSKeyedArchiver : NSCoder {
     ///   - path:       The path of the file in which to write the archive.
     /// - Returns:      `true` if the operation was successful, otherwise `false`.
     open class func archiveRootObject(_ rootObject: Any, toFile path: String) -> Bool {
+        var filePath = "/" + path
```

Comment:
Path input like "Users/sdsfd/test.txt" is accepted on Darwin and a "/" is prepended to the path. Hence, prepending the path since a double "/" will not impact.

Reply:
Is this the only normalisation that Darwin does? It seems unlikely to me... I would guess that there is more to it than this. Also, why not only prepend if required - allowing a double // seems a bit hacky.

Reply:
I don't think prepending with `/` is valid here. If I try to write to a relative path (say, `"./my-archive"`), we'd be making it absolute (`"/./my-archive"`), which is not correct.

Why is the current behavior a failure on Linux? What paths are failing? Note that in Darwin Foundation, we don't modify this path at all; the underlying writing code validates the path and adds a `/` if necessary.